app/DTO/PhotoCreateDTO.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\DTO;
+
+use App\Contracts\Image\ImageHandlerInterface;
+use App\Contracts\Image\StreamStats;
+use App\Contracts\Models\AbstractAlbum;
+use App\Contracts\Models\AbstractSizeVariantNamingStrategy;
+use App\Exceptions\Internal\LycheeLogicException;
+use App\Image\Files\BaseMediaFile;
+use App\Image\Files\FlysystemFile;
+use App\Image\Files\NativeLocalFile;
+use App\Image\Files\TemporaryLocalFile;
+use App\Metadata\Extractor;
+use App\Models\Photo;
+
+class PhotoCreateDTO
```

Comment:
This feels like an extension of `InitDTO`. I don't know enough to know whether that would be better, worse or impossible.

Reply:
Good catch, that one is a mistake in the commit. :')

It was a god object in PR #2253 
now I am using DTO specific to the pipes being applied: Init in the first pipes, Duplicates in case of duplication see #2369 etc.

Reply:
Ah

Reply:
Not sure whether that's for now or later, so I'll approve this and you can decide :)

Reply:
Fixed [3153680](https://github.com/LycheeOrg/Lychee/pull/2368/commits/31536800ec1c9db6344b754855431e6cac308071)